### ISSUE TITLE
Support react 16.x

### ISSUE THREAD

**Issue by evisong** [mobxjs/mobx-react#233]
Hi, 

Just found that React-Native 0.43 now depends on react 16.x (react@16.0.0-alpha.6), any plan for mobx-react to support react 16.x?

Thanks.

**Comment by mweststrate** [comment]
Afaik nothing should break with react@16.0, if you are interested, feel free to open a PR that bumps the dev and peer deps. If the testsuite succeeds we should be good to go :)

**Comment by evisong** [comment]
Hi, I've tried to upgrade to latest `react@16.0.0-alpha.7` (FYI it's `react-native@0.43`'s dependency) but failed due to lack of corresponding version for `react-addons-test-utils`, whose latest published version is `react-addons-test-utils@16.0.0-alpha.3`. So I end up with fixed versions:

- `react@16.0.0-alpha.3`
- `react-addons-test-utils@16.0.0-alpha.3`
- `react-dom@16.0.0-alpha.3`

Above code changes have been pushed to https://github.com/evisong/mobx-react/tree/issue_233_support_react_16

Unfortunately, `npm run test:browser` fails in the following test suites:

```
not ok 30 should be equal
  ---
    operator: equal
    expected: |-
      'child:3 - parent:3'
    actual: |-
      'child:3 - parent:2'
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 31 should be equal
  ---
    operator: equal
    expected: |-
      'child:5 - parent:5'
    actual: |-
      'child:3 - parent:2'
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 32 should be equal
  ---
    operator: equal
    expected: |-
      'child:7 - parent:7'
    actual: |-
      'child:3 - parent:2'
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 83 renderCount === 2
  ---
    operator: equal
    expected: 2
    actual:   1
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 84 renderCount === 3
  ---
    operator: equal
    expected: 3
    actual:   1
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 99 should be equal
  ---
    operator: equal
    expected: 'hello'
    actual:   'hi'
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...
not ok 107 should be equal
  ---
    operator: equal
    expected: 3
    actual:   2
    at: Test.assert [as _assert] (file:///Users/evisong/dev/projects/evisong/evisong-mobx-react/test/browser/test_bundle.js:101803:17)
  ...

1..279
# tests 279
# pass  272
# fail  7
```

I guess those breaks are caused by React Fiber introduced in react@16.0.0 (https://github.com/acdlite/react-fiber-architecture), but I'm sure since it's still in alpha phase. 

Do you suggest we wait for release version of react@16.0.0, or follow up as early as possible?

Thanks.

**Comment by iotch** [comment]
> Afaik nothing should break with react@16.0

Mobx-react will not work with React v16 without rewriting it to es6 classes or use standalone `prop-types` and `create-react-class` packages. Currently, with React v15.5.x, mobx-react is already polluting console with deprecation warnings.

**Comment by mweststrate** [comment]
Fixed in 4.1.6

**Comment by andykog** [comment]
Correction: fixed in 4.1.7
